### PR TITLE
docs: block file format ADR

### PR DESCRIPTION
**Pull request by melekes**
Refs #4630

### CODE REVIEW

File: docs/architecture/adr-060-block-file-format.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+to one stored in the database.
+
+```go
+type BlockMeta struct {
+	BlockID        BlockID  `json:"block_id"`
+	BlockSize      int      `json:"block_size"`
+	Header         Header   `json:"header"`
+	NumTxs         int      `json:"num_txs"`
+	Checksum       uint32   `json:"checksum"`
+	PartsChecksums []uint32 `json:"parts_checksums"`
```
Comment (erikgrinaker) [inline docs/architecture/adr-060-block-file-format.md:29]: Should this be `PartOffsets`? From the description below, it sounds like these are the file offsets of the parts, not checksums.

**Comment by marbar3778** [comment]
How or will this cause changes for the error "too many files open"? Will users need to increase the allowed open file setting for the amount of blocks the chain has?

**Comment by melekes** [comment]
> How or will this cause changes for the error "too many files open"? Will users need to increase the allowed open file setting for the amount of blocks the chain has?

It will reduce the number of files since we'll be storing less data in GoLevelDB, which is the reason we have many open files.

**Comment by melekes** [comment]
@erikgrinaker have suggested using BTree (boltdb or our tree?) if we need fast access to block parts (when helping other peers to catch up). Need to weight all benefits / downsides. 

Plus it will be helpful to perform benchmarks on real world data.

File: docs/architecture/adr-060-block-file-format.md (hunk 1 above)
Comment (melekes) [reply docs/architecture/adr-060-block-file-format.md:29]: We don't need offsets since part size is always fixed (even with erasure coding).

**Comment by erikgrinaker** [comment]
>"too many files open"

This is generally caused by networking, not actual files. In UNIX, everything is a file, including a network connection - so if you have 256 connections to/from peers, you're using 256 file handles.

**Comment by erikgrinaker** [comment]
> boltdb or our tree?

BoltDB, for sure

**Comment by marbar3778** [comment]
> BoltDB, for sure. And the main motivation is to avoid compaction pauses, which is the main performance issue we're facing from what I can tell.

We do only use the default settings for DBs, there are better configurations we should take advantage of. The first one that comes to mind is: https://github.com/syndtr/goleveldb/pull/297.

**Pull request closed without merging**